source/grid/grid_generator_from_name.cc

```diff
@@ -0,0 +1,331 @@
+    {
+      return false;
+    }
+
+    /**
+     * Methods implemented only for Triangulation<3,3>.
+     */
+    bool
+    generate_special(const std::string &  name,
+                     const std::string &  arguments,
+                     Triangulation<3, 3> &tria)
+    {
+      if (name == "moebius")
+        parse_and_create<3, 3, unsigned int, unsigned int, double, double>(
+          moebius, arguments, tria);
+      else if (name == "torus")
+        parse_and_create<3, 3, double, double, unsigned int>(torus,
+                                                             arguments,
+                                                             tria);
+      else
+        {
+          return false;
+        }
+      return true;
+    }
+
+    /**
+     * Methods implemented only for Triangulation<2,3>.
+     */
+    bool
+    generate_special(const std::string &  name,
+                     const std::string &  arguments,
+                     Triangulation<2, 3> &tria)
+    {
+      if (name == "torus")
+        parse_and_create<2, 3, double, double, unsigned int>(torus,
+                                                             arguments,
+                                                             tria);
+      else
+        {
+          return false;
+        }
+      return true;
+    }
+  } // namespace
+
+  template <int dim, int spacedim>
+  void
+  generate_from_name_and_arguments(Triangulation<dim, spacedim> &tria,
+                                   const std::string &           name,
+                                   const std::string &           arguments)
+  {
+    if (name == "hyper_cube")
+      parse_and_create<dim, spacedim, double, double, bool>(hyper_cube,
+                                                            arguments,
+                                                            tria);
+    else if (name == "subdivided_hyper_cube")
+      parse_and_create<dim, spacedim, unsigned int, double, double>(
+        subdivided_hyper_cube, arguments, tria);
+    else if (name == "hyper_rectangle")
+      parse_and_create<dim,
+                       spacedim,
+                       const Point<dim> &,
+                       const Point<dim> &,
+                       bool>(hyper_rectangle, arguments, tria);
+    else if (name == "cheese")
+      parse_and_create<dim, spacedim, const std::vector<unsigned int> &>(
+        cheese, arguments, tria);
+    else if (name == "general_cell")
+      parse_and_create<dim,
+                       spacedim,
+                       const std::vector<Point<spacedim>> &,
+                       bool>(general_cell, arguments, tria);
+    else if (name == "hyper_cross")
+      parse_and_create<dim, spacedim, const std::vector<unsigned int> &, bool>(
+        hyper_cross, arguments, tria);
+    else if (generate_codimension_zero_grid(name, arguments, tria))
+      {}
+    else if (generate_codimension_one_grid(name, arguments, tria))
```

Comment:
Ha, funny sequence here. I think a comment of the form "Now try each of the following options one after the other and see which one works" would be helpful.